source/megengine.js

```diff
@@ -8,20 +8,16 @@ megengine.ModelFactory = class {
 
     match(context) {
         const stream = context.stream;
-        if (stream && stream.length >= 12) {
-            const buffer = stream.peek(12);
-            const size = buffer[0] + (buffer[1] << 8) + (buffer[2] << 16) + (buffer[3] << 24);
-            if (size === (stream.length - 4)) {
+        if (stream && stream.length >= 24) {
+            var i = String(stream.peek(24)).indexOf("mgv2");
+            if (i >= 0) {
+                i = ((i - 8) > 0 ? (i - 8): 0);
+                stream.skip(i);
+                const buffer = stream.peek(12);
                 const reader = flatbuffers.BinaryReader.open(buffer.slice(4, 12));
                 if (reader.identifier === 'mgv2') {
-                    return 'megengine.mge';
-                }
+                    return 'megengine.flatbuffers';
             }
-            const tag = String.fromCharCode.apply(null, buffer);
-            for (const value of [ 'mgb0001', 'mgb0000a', 'MGBS', 'MGBC', 'mgbtest0' ]) {
-                if (tag.startsWith(value)) {
-                    return 'megengine.' + value;
-                }
```

Comment:
Why remove the code detecting other formats?

Reply:
👍👍 Sorry, it was deleted by mistake.

Reply:
Why is this code change needed? Is there a repro example?

```javascript
var i = String(stream.peek(24)).indexOf("mgv2");
if (i >= 0) {
    i = ((i - 8) > 0 ? (i - 8): 0);
    stream.skip(i);
}
```

Reply:
Because sometimes some leading information may be added before calling dump in python, although this is not a normal usage. For example:
$ hexdump -C  -n 100  model.mge 
00000000  6d 67 62 74 65 73 74 30  01 00 00 00 54 b8 16 08  |mgbtest0....T...|
00000010  1c 00 00 00 6d 67 76 32  14 00 24 00 20 00 1c 00  |....mgv2..$. 
```
            const tag = String.fromCharCode.apply(null, buffer);
            for (const value of [ 'mgb0001', 'mgb0000a', 'MGBS', 'MGBC', 'mgbtest0' ]) {
                if (tag.startsWith(value)) {
                    return 'megengine.' + value;
                }
```
I found that the above code was added by you not long ago. Some models do have this information at the beginning, but they use a non fbs serialization method.

Reply:
Does the added information have a standardized format that can be detected? What if the information is more than 24 characters or changes in the future? Can you share or add sample test files.

Reply:
There seems to be no standard format for the information added. The added information is actually written in before calling the serialized interface (`dump`), so it won't exist if you just call the serialized interface (`dump`) and don't write other information before.

For the mge model format that can be visualized by netron, there is only one possible addition of information: mgbtest0. The added information is used to represent the presence of user-given input data in the model and can be identified by the tools (mge/load_and_run) that we provide to the user to analyze the inference performance of the model.

In general, model serialization should only be model serialization, but in mge, these short lead information were added to accommodate the use of other tools. So if it's more than 24 bytes, I think we should modify the way our tool recognizes it?

Reply:
Is `mgbtest0` a known header format? For example, detect, read its length, skip over it?

Reply:
Yes, there will be a fixed 12-Byte offset. Fixed in this way.